v2/orderedmap.go

```diff
@@ -152,3 +155,28 @@ func (m *OrderedMap[K, V]) Copy() *OrderedMap[K, V] {
 
 	return m2
 }
+
+// Implements JSON marshaling.
+func (m *OrderedMap[string, V]) MarshalJSON() ([]byte, error) {
```

Comment:
I'm not sure how I feel about this for a few reasons:

1. It only works with `string` keys. Which I'm sure is a common case, but I don't want to get into needing to do a method for every common type.
2. It's not symmetrical (there is no decode/unmarshal).
3. It's quite opinionated in the way that it's doing the encode. Specifically, it creates a JSON object that is ambiguous because _some_ systems that ingest this ordered data (for example, Go itself) which will not reasonably be able to restore the order so information will be lost.

I think a more appropriate way to deal with this is to have separate symmetrical types that can wrap depending on the functionality needed (basically what users have to do now) like:

```go
type MyThingo struct {
     // Great for memory, but not able to serialize.
    map1 *orderedmap.OrderedMap[string, any]

    // I need to serialize this (and maintain order), but don't care
    // about the storage format. Like:
    //   [{"key":"foo","value":123}]
    map2 *orderedmap.OrderedSerializableMap[string, any]

    // I want to retain the order in memory, but I don't need to maintain
    // when when serialized/reduce foot print/increase readability by
    // other programs (probably a bad name for the type though). Like:
    //   {"foo",123}
    map3 *orderedmap.OrderedMapSerializeUnordered[string, any]
}